sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/InjectRuntimeFilter.scala

```diff
@@ -99,7 +99,7 @@ object InjectRuntimeFilter extends Rule[LogicalPlan] with PredicateHelper with J
     require(filterApplicationSideExp.dataType == filterCreationSideExp.dataType)
     val actualFilterKeyExpr = mayWrapWithHash(filterCreationSideExp)
     val alias = Alias(actualFilterKeyExpr, actualFilterKeyExpr.toString)()
-    val aggregate = Aggregate(Seq(alias), Seq(alias), filterCreationSidePlan)
+    val aggregate = ColumnPruning(Aggregate(Seq(alias), Seq(alias), filterCreationSidePlan))
```

Comment:
This looks a bit hacky. Can we use the catalyst framework to optimize it?

Reply:
I'm afraid that will be overkill to apply the whole optimizer rules. The `filterCreationSidePlan` is simple enough, it can only contain project and filter. Apply ColumnPruning here seems more safer ?

Reply:
how about the filter push rule?

Reply:
The `filterCreationSidePlan` has been optimized , so it should be done if there is a filter can be pushed.

I think one more useful rule is `CollapseProject`, but it should be fine not to apply here since `PhysicalOperation` support collect adjacent projects.